bindings/azure/cosmosgraphdb/cosmosgraphdb.go

```diff
@@ -0,0 +1,130 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package cosmosgraphdb
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+	gremcos "github.com/supplyon/gremcos"
+)
+
+const (
+	queryOperation bindings.OperationKind = "query"
+
+	// keys from request's metadata
+	commandGremlinKey = "gremlin"
+
+	// keys from response's metadata
+	respGremlinKey   = "gremlin"
+	respOpKey        = "operation"
+	respStartTimeKey = "start-time"
+	respEndTimeKey   = "end-time"
+	respDurationKey  = "duration"
+)
+
+// CosmosGraphDB allows performing state operations on collections
+type CosmosGraphDB struct {
+	metadata *cosmosGraphDBCredentials
+	client   *gremcos.Cosmos
+	logger   logger.Logger
+}
+
+type cosmosGraphDBCredentials struct {
+	URL       string `json:"url"`
+	MasterKey string `json:"masterKey"`
+	Username  string `json:"username"`
+	// NumMaxActiveConnections string `json:"NumMaxActiveConnections"`
+	// ConnectionIdleTimeout   string `json:"ConnectionIdleTimeout"`
```

Comment:
Do we not want to use these and give defaults if they aren't provided?

Reply:
yes, @halspang you are right , some comments need to be removed along with these values (NumMaxActiveConnections, ConnectionIdleTimeout)  , I felt like it is over complicating the component YAML file , so I decided to remove it . any way the gremcos package assign there values to a default values which is more than enough for now.

Reply:
All unwanted comment removed.